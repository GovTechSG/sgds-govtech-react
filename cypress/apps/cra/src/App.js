import 'bootstrap-icons/font/bootstrap-icons.css';
import '@govtechsg/sgds/sass/sgds.scss';
import '@govtechsg/sgds-masthead/dist/sgds-masthead/sgds-masthead.css';

import {
  FileUploadCom,
  MastheadCom,
  TooltipCom,
  NavbarCom,
  StepperCom,
  AccordionCom,
  AlertCom,
  BadgeCom,
  BreadcrumbCom,
  CardCom,
  QtyToggleCom,
  SelectableCardCom,
  DropdownCom,
  ValidationCom,
  InputGroupCom,
  TableCom,
  ToastCom,
  PaginationCom,
  SideNavCom,
  ProgressBarCom,
  DatePickerCom,
  ComboboxCom,
  ModalCom,
  TabsCom
} from './components';

function App() {
  return (
    <div className="container">
      <MastheadCom />
      <FileUploadCom />
      <TooltipCom />
      <NavbarCom />
      <StepperCom />
      <AccordionCom />
      <AlertCom />
      <BadgeCom/>
      <BreadcrumbCom/>
      <CardCom/>
      <QtyToggleCom/>
      <SelectableCardCom/>
      <DropdownCom/>
      <ValidationCom/>
      <InputGroupCom/>
      <TableCom/>
      <ToastCom/> 
      <PaginationCom/>
      <SideNavCom/>
      <ProgressBarCom/> 
      <DatePickerCom/>
      <ComboboxCom/>
      <ModalCom/>
      <TabsCom/>
    </div>
  );
}

export default App;
