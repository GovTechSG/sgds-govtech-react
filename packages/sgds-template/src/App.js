import { BrowserRouter as Router, Route, Routes } from "react-router-dom";
import { NavbarTemp } from "./components/NavbarTemp";
import { SinglePageFormTemp } from "./components/SinglePageFormTemp";
import { MultiPageFormTemp } from "./components/MultiPageFormTemp";
import { CollapsibleFormTemp } from "./components/CollapsibleFormTemp";
import "@govtechsg/sgds-govtech/sgds/sgds.css";

function App() {
  return (
    <Router>
      <div className='mb-3'>
        <NavbarTemp />
      </div>
      <div className='mb-3'>
        <Routes>
          <Route path='/' element={<SinglePageFormTemp />} />
          <Route path='/multi' element={<MultiPageFormTemp />} />
          <Route path='/collapsible' element={<CollapsibleFormTemp />} />
        </Routes>
      </div>
    </Router>
  );
}

export default App;
