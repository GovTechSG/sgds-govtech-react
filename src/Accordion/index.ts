export { default as Accordion } from './Accordion';
export type { AccordionProps } from './Accordion';

export { default as AccordionContext } from './AccordionContext';
export { default as AccordionCollapse } from './AccordionCollapse';
export type { AccordionCollapseProps } from './AccordionCollapse';

export {
  default as AccordionButton,
  useAccordionButton,
} from './AccordionButton';
export type { AccordionButtonProps } from './AccordionButton';

export { default as AccordionItem } from './AccordionItem';
export type { AccordionItemProps } from './AccordionItem';
export { default as AccordionHeader } from './AccordionHeader';
export type { AccordionHeaderProps } from './AccordionHeader';
export { default as AccordionBody } from './AccordionBody';
export type { AccordionBodyProps } from './AccordionBody';
