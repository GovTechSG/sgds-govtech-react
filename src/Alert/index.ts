export { default as Alert } from './Alert';
export type { AlertProps } from './Alert';

export { default as AlertLink } from './AlertLink';
// export type { AlertLinkProps } from './AlertLink';

export { default as AlertHeading } from './AlertHeading';
