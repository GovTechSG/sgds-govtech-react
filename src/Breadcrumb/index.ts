export { default as Breadcrumb } from './Breadcrumb';
export type { BreadcrumbProps } from './Breadcrumb';

export { default as BreadcrumbItem } from './BreadcrumbItem';
export type { BreadcrumbItemProps } from './BreadcrumbItem';