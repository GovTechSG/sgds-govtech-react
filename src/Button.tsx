// import React, { HTMLAttributes, ReactNode } from 'react';

// export interface Props extends HTMLAttributes<HTMLButtonElement> {
// /** Provide a text for the button */
//   children: ReactNode;
// /** Which variant you would like to use */
//   variant: 'primary' | 'secondary';
// }

// /** this is a special button */
// export const Button = ({ children, variant = 'primary', ...props }: Props) => {
//   return (
//     <button
//       {...props}
//       style={{
//         backgroundColor: variant === 'primary' ? 'blue' : 'gray',
//         color: 'white',
//         border: 'none',
//         borderRadius: '100px',
//         padding: '10px',
//         cursor: 'pointer'
//       }}
//     >
//       {children}
//     </button>
//   );
// };
