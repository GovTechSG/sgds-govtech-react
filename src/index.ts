export * from './Accordion'
// export { default as Accordion } from './Accordion/Accordion';
// export type { AccordionProps } from './Accordion/Accordion';

// export { default as AccordionContext } from './Accordion/AccordionContext';
// export { default as AccordionCollapse } from './Accordion/AccordionCollapse';
// export type { AccordionCollapseProps } from './Accordion/AccordionCollapse';

// export {
//   default as AccordionButton,
//   useAccordionButton,
// } from './Accordion/AccordionButton';
// export type { AccordionButtonProps } from './Accordion/AccordionButton';


export * from './Alert'
// export { default as Alert } from './Alert/Alert';
// export type { AlertProps } from './Alert/Alert';

// export { default as Anchor } from './Anchor';
// export type { AnchorProps } from './Anchor';
export * from './Badge'
// export { Badge } from './Badge/Badge';
// export type { BadgeProps } from './Badge/Badge';
export * from './Breadcrumb'
// export { default as Breadcrumb } from './Breadcrumb/Breadcrumb';
// export type { BreadcrumbProps } from './Breadcrumb/Breadcrumb';

// export { default as BreadcrumbItem } from './Breadcrumb/BreadcrumbItem';
// export type { BreadcrumbItemProps } from './Breadcrumb/BreadcrumbItem';

// export { default as Button } from './Button/Button';
// export type { ButtonProps } from './Button/Button';
export * from './Button'
export * from './ButtonGroup'
// export { default as ButtonGroup } from './ButtonGroup';
// export type { ButtonGroupProps } from './ButtonGroup';

// export { default as ButtonToolbar } from './ButtonToolbar';
// export type { ButtonToolbarProps } from './ButtonToolbar';
export * from './Card'
//   export { default as Card } from './Card/Card';
// export type { CardProps } from './Card/Card';
// export { default as SelectableCard } from './Card/SelectableCard'
// export type { SelectableCardProps } from './Card/SelectableCard'

// export { default as CardImg } from './Card/CardImg';
// export type { CardImgProps } from './Card/CardImg';

// export { default as CardGroup } from './Card/CardGroup';
// export { default as Carousel } from './Carousel';
// export type { CarouselProps } from './Carousel';

// export { default as CarouselItem } from './CarouselItem';
// export type { CarouselItemProps } from './CarouselItem';
export * from './CloseButton'
// export { default as CloseButton } from './CloseButton/CloseButton';
// export type { CloseButtonProps } from './CloseButton/CloseButton';
export * from './Col'
// export { default as Col } from '../Col';
// export type { ColProps } from '../Col';
export * from './CloseButton'
export * from './Collapse'
// export { default as Collapse } from './Collapse/Collapse';
// export type { CollapseProps } from './Collapse/Collapse';
export * from './Dropdown'
// export { default as Dropdown } from './Dropdown/Dropdown';
// export type { DropdownProps } from './Dropdown/Dropdown';

// export { default as DropdownButton } from './Dropdown/DropdownButton';
// export type { DropdownButtonProps } from './Dropdown/DropdownButton';
export * from './Fade'
// export { default as Fade } from './Fade/Fade';
// export type { FadeProps } from './Fade/Fade';
export * from './Form'
// export { default as Form } from './Form/Form';
// export type { FormProps } from './Form/Form';

// export { default as FormControl } from './Form/FormControl';
// export type { FormControlProps } from './Form/FormControl';

// export { default as FormCheck } from './Form/FormCheck';
// export type { FormCheckProps } from './Form/FormCheck';

// export { default as FormFloating } from './Form/FormFloating';
export * from './FloatingLabel'
// export { default as FloatingLabel } from '../FloatingLabel';
// export type { FloatingLabelProps } from '../FloatingLabel';

// export { default as FormGroup } from './Form/FormGroup';
// export type { FormGroupProps } from './Form/FormGroup';

// export { default as FormLabel } from './Form/FormLabel';
// export type { FormLabelProps } from './Form/FormLabel';

// export { default as FormText } from './Form/FormText';
// export type { FormTextProps } from './Form/FormText';

// export { default as FormSelect } from './Form/FormSelect';
// export type { FormSelectProps } from './Form/FormSelect';

// export { default as Container } from './Container';
// export type { ContainerProps } from './Container';

// export { default as Image } from './Image';
// export type { ImageProps } from './Image';

// export { default as Figure } from './Figure';
export * from './InputGroup'
// export { default as InputGroup } from './InputGroup/InputGroup';
// export type { InputGroupProps } from './InputGroup/InputGroup';

// export { default as ListGroup } from './ListGroup';
// export type { ListGroupProps } from './ListGroup';

// export { default as ListGroupItem } from './ListGroupItem';
// export type { ListGroupItemProps } from './ListGroupItem';
export * from './Modal'
// export { default as Modal } from './Modal/Modal';
// export type { ModalProps } from './Modal/Modal';

// export { default as ModalBody } from './Modal/ModalBody';
// export { default as ModalDialog } from './Modal/ModalDialog';
// export type { ModalDialogProps } from './Modal/ModalDialog';

// export { default as ModalFooter } from './Modal/ModalFooter';
// export { default as ModalTitle } from './Modal/ModalTitle';
export * from './Nav'
// export { default as Nav } from '../Nav/Nav';
// export type { NavProps } from '../Nav/Nav';

// // export { default as Navbar } from './Navbar';
// // export type { NavbarProps } from './Navbar';

// // export { default as NavbarBrand } from './NavbarBrand';
// // export type { NavbarBrandProps } from './NavbarBrand';

// // export { default as NavDropdown } from './NavDropdown';
// // export type { NavDropdownProps } from './NavDropdown';

// export { default as NavItem } from './Nav/NavItem';

// export { default as NavLink } from './Nav/NavLink';
// export type { NavLinkProps } from './Nav/NavLink';

// export { default as Offcanvas } from './Offcanvas';
// export type { OffcanvasProps } from './Offcanvas';
// export { default as OffcanvasHeader } from './OffcanvasHeader';
// export type { OffcanvasHeaderProps } from './OffcanvasHeader';
// export { default as OffcanvasTitle } from './OffcanvasTitle';
// export { default as OffcanvasBody } from './OffcanvasBody';
export * from './Overlay'
// export { default as Overlay } from './Overlay/Overlay';
// export type { OverlayProps } from './Overlay/Overlay';

// export { default as OverlayTrigger } from './Overlay/OverlayTrigger';
// export type { OverlayTriggerProps } from './Overlay/OverlayTrigger';
export * from './PaginationBase'
// export { default as PageItem } from './PaginationBase/PageItem';
// export type { PageItemProps } from './PaginationBase/PageItem';

// export { default as PaginationBase } from './PaginationBase/PaginationBase';
// export type { PaginationBaseProps } from './PaginationBase/PaginationBase';
export * from './Paginations'
// export { Paginations } from './Paginations/Paginations';
// export type { PaginationsProps } from './Paginations/Paginations';

// export { default as Placeholder } from './Placeholder';
// export type { PlaceholderProps } from './Placeholder';
// export { default as PlaceholderButton } from './PlaceholderButton';
// export type { PlaceholderButtonProps } from './PlaceholderButton';
export * from './Popover'
// export { default as Popover } from './Popover';
// export type { PopoverProps } from './Popover';

// export { default as PopoverHeader } from './PopoverHeader';
// export { default as PopoverBody } from './PopoverBody';
export * from './ProgressBar'
// export { default as ProgressBar } from './ProgressBar/ProgressBar';
// export type { ProgressBarProps } from './ProgressBar/ProgressBar';
export * from './QuantityToggle'
// export {default as QuantityToggle} from './QuantityToggle/QuantityToggle'
// export type { QuantityToggleProps } from './QuantityToggle/QuantityToggle'
// export { default as Ratio } from './Ratio';
// export type { RatioProps } from './Ratio';
export * from './Row' 
// export { default as Row } from '../Row';
// export type { RowProps } from '../Row';
export * from './SideNav'
// export { default as SideNav } from './SideNav/SideNav';
// export type { SideNavProps } from './SideNav/SideNav';

// export { default as Spinner } from './Spinner';
// export type { SpinnerProps } from './Spinner';


export * from './SplitButton'
// export { default as SplitButton } from '../SplitButton/SplitButton';
// export type { SplitButtonProps } from '../SplitButton/SplitButton';

// export { default as SSRProvider } from './SSRProvider';
// export type { SSRProviderProps } from './SSRProvider';

// export { default as Stack } from './Stack';
// export type { StackProps } from './Stack';
export * from './Stepper'
// export { default as Stepper } from './Stepper/Stepper'
// export { default as useStep } from './Stepper/useStep'
// export type { OnDone, StepMetadata, State as StepperState, UseStepMethods} from './Stepper/useStep'

// export { default as Tab } from './Tabs/Tab';
// export type { TabProps } from './Tabs/Tab';

// export { default as TabContainer } from './Tabs/TabContainer';
// export type { TabContainerProps } from './Tabs/TabContainer';

// export { default as TabContent } from './Tabs/TabContent';
export * from './Table'
// export { default as Table } from './Table/Table';
// export type { TableProps } from './Table/Table';

// export { default as TabPane } from './Tabs/TabPane';
// export type { TabPaneProps } from './Tabs/TabPane';

// export { default as Tabs } from './Tabs/Tabs';
// export type { TabsProps } from './Tabs/Tabs';
export * from './ThemeProvider'
// export { default as ThemeProvider } from './ThemeProvider/ThemeProvider';
// export type { ThemeProviderProps } from './ThemeProvider/ThemeProvider';
export * from './Toast'
// export { default as Toast } from './Toast/Toast';
// export type { ToastProps } from './Toast/Toast';

// export { default as ToastBody } from './Toast/ToastBody';
// export { default as ToastHeader } from './Toast/ToastHeader';
// export type { ToastHeaderProps } from './Toast/ToastHeader';

// export { default as ToastContainer } from './Toast/ToastContainer';
// export type { ToastContainerProps } from './Toast/ToastContainer';

// export { default as ToggleButton } from './ToggleButton';
// export type { ToggleButtonProps } from './ToggleButton';

// export { default as ToggleButtonGroup } from './ToggleButtonGroup';
// export type {
//   ToggleButtonCheckboxProps,
//   ToggleButtonGroupProps,
//   ToggleButtonRadioProps,
// } from './ToggleButtonGroup';
export * from './Tooltip'
// export { default as TooltipBox } from './Tooltip/TooltipBox';
// export type { TooltipBoxProps } from './Tooltip/TooltipBox';

// export { default as Tooltip } from './Tooltip/Tooltip';
// export type { TooltipProps } from './Tooltip/Tooltip';


