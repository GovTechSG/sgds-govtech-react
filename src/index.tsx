
export * from '../src/components/index'
// export * from '../src/components/Alert/Alert'
// export * from '../src/components/Button/Button'
// export * from '../src/components/Card/Card'
// export * from '../src/components/Dropdown/Dropdown'
// export * from '../src/components/Form/Form'
// export * from '../src/components/Modal/Modal'
// export * from '../src/components/Badge/Badge'
// export * from '../src/components/Breadcrumb/Breadcrumb'
// export * from '../src/components/Card/Card'
// export * from '../src/components/Card/CardImg'
// export * from '../src/components/Card/CardGroup'
// export * from '../src/components/Card/Card'
// export * from '../src/components/CloseButton/CloseButton'
// export * from '../src/components/Collapse/Collapse';
// export * from '../src/components/Dropdown/Dropdown';
// export * from '../src/components/Dropdown/DropdownButton';
// export * from '../src/components/Fade/Fade';
// export * from '../src/components/Form/Form';
// export * from '../src/components/Form/FormControl';
// export * from '../src/components/Form/FormCheck';
// export * from '../src/components/Form/FormGroup';
// export * from '../src/components/Form/FormLabel';
// export * from '../src/components/Form/FormText';
// export * from '../src/components/Form/FormSelect';
// export * from '../src/components/InputGroup/InputGroup';
// export * from '../src/components/Modal/Modal';
// export * from '../src/components/Modal/ModalBody';
// export * from '../src/components/Modal/ModalDialog';
// export * from '../src/components/Modal/ModalFooter';
// export * from '../src/components/Modal/ModalTitle';
// export * from '../src/components/Overlay/Overlay';
// export * from '../src/components/PaginationBase/PageItem';
// export * from '../src/components/PaginationBase/PaginationBase';
// export * from '../src/components/Paginations/Paginations';
// export * from '../src/components/ProgressBar/ProgressBar';
// export * from '../src/components/QuantityToggle/QuantityToggle'
// export * from '../src/components/Row';
// export * from '../src/components/SplitButton/SplitButton';
// export * from '../src/components/Stepper/Stepper'
// export * from '../src/components/Stepper/useStep'
// export * from '../src/components/Table/Table';
// export * from '../src/components/ThemeProvider/ThemeProvider';
// export * from '../src/components/Toast/Toast';
// export * from '../src/components/Toast/ToastBody';
// export * from '../src/components/Toast/ToastHeader';
// export * from '../src/components/Toast/ToastContainer';
// export * from '../src/components/Tooltip/TooltipBox';
// export * from '../src/components/Tooltip/Tooltip';