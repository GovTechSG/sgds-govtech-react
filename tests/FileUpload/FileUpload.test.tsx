import { render } from '@testing-library/react';
import * as React from 'react';

import { FileUpload } from '../../src';
interface iBlob extends Blob, File {
  name: string;
  lastModifiedDate: Date;
  lastModified: number;
  webkitRelativePathts: string;
}

interface MockFile {
  name: string;
  body: string;
  mimeType: string;
}
const createFileFromMockFile = (file: MockFile): File => {
  const blob: Partial<iBlob> = new Blob([file.body], { type: file.mimeType });
  blob['lastModifiedDate'] = new Date();
  blob['name'] = file.name;
  return blob as File;
};

const createMockFileList = (files: MockFile[]) => {
  const fileList: FileList = {
    length: files.length,
    item(index: number): File {
      return fileList[index];
    },
  };
  files.forEach(
    (file, index) => (fileList[index] = createFileFromMockFile(file))
  );

  return fileList;
};

describe('<FileUpload>', () => {
  const mockFn = jest.fn();

  it('should be able to pass in content in between tags', () => {
    const { getByText } = render(
      <FileUpload controlId="test123" onChangeFile={mockFn} selectedFile={{}}>
        FileUpload
      </FileUpload>
    );

    expect(getByText('FileUpload')).toBeDefined();
  });

  it('Should output a child input element', () => {
    const { asFragment, container } = render(
      <FileUpload
        data-testid="test"
        controlId="test123"
        onChangeFile={mockFn}
        selectedFile={{}}
      >
        FileUpload
      </FileUpload>
    );
    expect(asFragment()).toMatchSnapshot();
    expect(container.querySelector('div')).toBeDefined();
    const $fileUpload = container.querySelector('div');
    expect($fileUpload?.children.length).toEqual(2);
    expect($fileUpload?.children[0].tagName).toEqual('INPUT');

    const $input = $fileUpload?.children[0];
    expect($input).not.toHaveAttribute('multiple');
    expect($input).toHaveAttribute('type', 'file');
    expect($input).toHaveAttribute('id', 'test123');
    expect($input?.classList).toContain('form-control');

    // expect($fileUpload?.children[1].tagName).toEqual('BUTTON');
  });

  it('Should output a child button element', () => {
    const { asFragment, container } = render(
      <FileUpload
        data-testid="test"
        controlId="test123"
        onChangeFile={mockFn}
        selectedFile={{}}
      >
        FileUpload
      </FileUpload>
    );
    expect(asFragment()).toMatchSnapshot();
    expect(container.querySelector('div')).toBeDefined();
    const $fileUpload = container.querySelector('div');
    expect($fileUpload?.children.length).toEqual(2);
    expect($fileUpload?.children[1].tagName).toEqual('BUTTON');
    const $button = $fileUpload?.children[1];
    expect($button?.classList).toContain('btn');
    expect($button?.classList).toContain('btn-primary');
    expect($button?.classList).toContain('sgds');
    expect($button).toHaveAttribute('type', 'button');
  });

  //this
  it('should render a ul with list items, upon uploading', () => {
    const fileList = createMockFileList([
      {
        body: 'test',
        mimeType: 'text/plain',
        name: 'test.txt',
      },
      {
        body: 'test1',
        mimeType: 'text/plain',
        name: 'hello.txt',
      },
    ]);
    const { asFragment, container } = render(
      <FileUpload
        data-testid="test"
        controlId="test123"
        onChangeFile={mockFn}
        selectedFile={fileList}
      >
        FileUpload
      </FileUpload>
    );

    expect(asFragment()).toMatchSnapshot();
    const $listItems = container.querySelector('ul');
    expect($listItems).toBeDefined();
    expect($listItems?.firstElementChild?.children[1]).toHaveAttribute(
      'id',
      'test.txt'
    );
    expect($listItems?.children[1].children[1]).toHaveAttribute(
      'id',
      'hello.txt'
    );
    expect($listItems?.firstElementChild?.children[2]).toHaveAttribute(
      'aria-describedby',
      'test.txt'
    );
    expect($listItems?.firstElementChild?.children[2]).toHaveAttribute(
      'aria-label',
      'remove file'
    );
    expect($listItems?.firstChild).toHaveTextContent('test.txt');
    expect($listItems?.children[1]).toHaveTextContent('hello.txt');
  });

  //this
  it("should have li named 'test.txt' and not 'hello.txt', upon removeFile", async () => {
    const fileList = createMockFileList([
      {
        body: 'test',
        mimeType: 'text/plain',
        name: 'test.txt',
      },
      {
        body: 'test1',
        mimeType: 'text/plain',
        name: 'hello.txt',
      },
    ]);

    const { asFragment, container } = render(
      <FileUpload
        data-testid="test"
        controlId="test123"
        onChangeFile={mockFn}
        selectedFile={fileList}
      >
        FileUpload
      </FileUpload>
    );

    expect(asFragment()).toMatchSnapshot();
    const $lists = container.querySelectorAll('li');
    const firstItem = $lists[0];
    expect(firstItem.textContent).toBe('test.txt');
    expect(firstItem.querySelector('i.bi-x-circle')).toBeInTheDocument();
    // TODO:
    // fireEvent.click(firstItem.children[2]);

    // await waitFor(() => {
    //   expect(firstItem.textContent).toBe('hello.txt');
    // });

    // const $listItems = getAllByRole('listitem');
    // fireEvent.click($listItems[0].children[2], fileList);
  });

  //this
  it('check if cancelIcon and checkedIcon changes default icon', () => {
    const fileList = createMockFileList([
      {
        body: 'test',
        mimeType: 'text/plain',
        name: 'test.txt',
      },
      {
        body: 'test1',
        mimeType: 'text/plain',
        name: 'hello.txt',
      },
    ]);
    const { asFragment, container } = render(
      <FileUpload
        data-testid="test"
        controlId="test123"
        onChangeFile={mockFn}
        selectedFile={fileList}
        cancelIcon={<i className="bi bi-x-octagon"></i>}
        checkedIcon={<i className="bi bi-check2-circle"></i>}
      >
        FileUpload
      </FileUpload>
    );
    const defaultCancelIcons = container.querySelectorAll(
      '.bi.bi-x-circle.x-circle-icon'
    );
    expect(asFragment()).toMatchSnapshot();
    const $cancelIcons = container.querySelectorAll('.bi.bi-x-octagon');
    expect($cancelIcons).toBeDefined();
    expect(defaultCancelIcons.length).toBe(0);

    const defaultCheckedIcons = container.querySelectorAll(
      '.bi-check-lg.check-icon'
    );
    const $checkedIcons = container.querySelectorAll('.bi-check2-circle');
    expect($checkedIcons).toBeDefined();
    expect(defaultCheckedIcons.length).toBe(0);
  });

  it('when multiple prop is true input element should have multiple attribute', () => {

    const { container } = render(
      <FileUpload
        data-testid="test"
        controlId="test123"
        onChangeFile={mockFn}
        selectedFile={{}}
        multiple
      >
        FileUpload
      </FileUpload>
    );

    expect(container.querySelector('input')).toHaveAttribute("multiple")
  })

  it("accept prop when defined is forwarded to input element", async() => {
    const { container } = render(
      <FileUpload
        data-testid="test"
        controlId="test123"
        onChangeFile={mockFn}
        selectedFile={{}}
        accept=".png"
      >
        FileUpload
      </FileUpload>
    );
    expect(container.querySelector('input')).toHaveAttribute("accept", ".png")
  })
  
  it("buttonClassName prop is forwarded to the Button component of Fileupload", () => {
    const { container } = render(
      <FileUpload
        data-testid="test"
        controlId="test123"
        onChangeFile={mockFn}
        selectedFile={{}}
        buttonClassName="test-css-selector"
      >
        FileUpload
      </FileUpload>
    );
    expect(container.querySelector("button.sgds")).toHaveClass("test-css-selector")
  })
});

// wrap cancelIcon with a button element as cancelIcon is supposed to be a button
// add id on span element that displays filename
// add aria-describedby on the button element wrapping cancelIcon which points to the span element filename
// add aria-label on the button element to tell users that the button is cancelled
