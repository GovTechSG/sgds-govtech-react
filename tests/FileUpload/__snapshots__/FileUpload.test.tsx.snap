// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FileUpload> Should output a child button element 1`] = `
<DocumentFragment>
  <div>
    <input
      class="d-none form-control"
      id="test123"
      type="file"
    />
    <button
      class="btn btn-primary sgds"
      type="button"
    >
      FileUpload
    </button>
  </div>
  <ul
    class="fileupload-list sgds"
  />
</DocumentFragment>
`;

exports[`<FileUpload> Should output a child input element 1`] = `
<DocumentFragment>
  <div>
    <input
      class="d-none form-control"
      id="test123"
      type="file"
    />
    <button
      class="btn btn-primary sgds"
      type="button"
    >
      FileUpload
    </button>
  </div>
  <ul
    class="fileupload-list sgds"
  />
</DocumentFragment>
`;

exports[`<FileUpload> check if cancelIcon and checkedIcon changes default icon 1`] = `
<DocumentFragment>
  <div>
    <input
      class="d-none form-control"
      id="test123"
      type="file"
    />
    <button
      class="btn btn-primary sgds"
      type="button"
    >
      FileUpload
    </button>
  </div>
  <ul
    class="fileupload-list sgds"
  >
    <li
      class="fileupload-list-item"
    >
      <i
        class="bi bi-check2-circle me-2"
      />
      <span
        class="filename"
        id="test.txt"
      >
        test.txt
      </span>
      <button
        aria-describedby="test.txt"
        aria-label="remove file"
        class="bg-transparent border-0 ms-2"
      >
        <i
          class="bi bi-x-octagon"
        />
      </button>
    </li>
    <li
      class="fileupload-list-item"
    >
      <i
        class="bi bi-check2-circle me-2"
      />
      <span
        class="filename"
        id="hello.txt"
      >
        hello.txt
      </span>
      <button
        aria-describedby="hello.txt"
        aria-label="remove file"
        class="bg-transparent border-0 ms-2"
      >
        <i
          class="bi bi-x-octagon"
        />
      </button>
    </li>
    <li
      class="fileupload-list-item"
    >
      <i
        class="bi bi-check2-circle me-2"
      />
      <span
        class="filename"
      />
      <button
        aria-label="remove file"
        class="bg-transparent border-0 ms-2"
      >
        <i
          class="bi bi-x-octagon"
        />
      </button>
    </li>
    <li
      class="fileupload-list-item"
    >
      <i
        class="bi bi-check2-circle me-2"
      />
      <span
        class="filename"
        id="item"
      >
        item
      </span>
      <button
        aria-describedby="item"
        aria-label="remove file"
        class="bg-transparent border-0 ms-2"
      >
        <i
          class="bi bi-x-octagon"
        />
      </button>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`<FileUpload> should have li named 'test.txt' and not 'hello.txt', upon removeFile 1`] = `
<DocumentFragment>
  <div>
    <input
      class="d-none form-control"
      id="test123"
      type="file"
    />
    <button
      class="btn btn-primary sgds"
      type="button"
    >
      FileUpload
    </button>
  </div>
  <ul
    class="fileupload-list sgds"
  >
    <li
      class="fileupload-list-item"
    >
      <i
        class="bi bi-check-lg check-icon me-2"
      />
      <span
        class="filename"
        id="test.txt"
      >
        test.txt
      </span>
      <button
        aria-describedby="test.txt"
        aria-label="remove file"
        class="bg-transparent border-0 ms-2"
      >
        <i
          class="bi bi-x-circle x-circle-icon"
        />
      </button>
    </li>
    <li
      class="fileupload-list-item"
    >
      <i
        class="bi bi-check-lg check-icon me-2"
      />
      <span
        class="filename"
        id="hello.txt"
      >
        hello.txt
      </span>
      <button
        aria-describedby="hello.txt"
        aria-label="remove file"
        class="bg-transparent border-0 ms-2"
      >
        <i
          class="bi bi-x-circle x-circle-icon"
        />
      </button>
    </li>
    <li
      class="fileupload-list-item"
    >
      <i
        class="bi bi-check-lg check-icon me-2"
      />
      <span
        class="filename"
      />
      <button
        aria-label="remove file"
        class="bg-transparent border-0 ms-2"
      >
        <i
          class="bi bi-x-circle x-circle-icon"
        />
      </button>
    </li>
    <li
      class="fileupload-list-item"
    >
      <i
        class="bi bi-check-lg check-icon me-2"
      />
      <span
        class="filename"
        id="item"
      >
        item
      </span>
      <button
        aria-describedby="item"
        aria-label="remove file"
        class="bg-transparent border-0 ms-2"
      >
        <i
          class="bi bi-x-circle x-circle-icon"
        />
      </button>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`<FileUpload> should render a ul with list items, upon uploading 1`] = `
<DocumentFragment>
  <div>
    <input
      class="d-none form-control"
      id="test123"
      type="file"
    />
    <button
      class="btn btn-primary sgds"
      type="button"
    >
      FileUpload
    </button>
  </div>
  <ul
    class="fileupload-list sgds"
  >
    <li
      class="fileupload-list-item"
    >
      <i
        class="bi bi-check-lg check-icon me-2"
      />
      <span
        class="filename"
        id="test.txt"
      >
        test.txt
      </span>
      <button
        aria-describedby="test.txt"
        aria-label="remove file"
        class="bg-transparent border-0 ms-2"
      >
        <i
          class="bi bi-x-circle x-circle-icon"
        />
      </button>
    </li>
    <li
      class="fileupload-list-item"
    >
      <i
        class="bi bi-check-lg check-icon me-2"
      />
      <span
        class="filename"
        id="hello.txt"
      >
        hello.txt
      </span>
      <button
        aria-describedby="hello.txt"
        aria-label="remove file"
        class="bg-transparent border-0 ms-2"
      >
        <i
          class="bi bi-x-circle x-circle-icon"
        />
      </button>
    </li>
    <li
      class="fileupload-list-item"
    >
      <i
        class="bi bi-check-lg check-icon me-2"
      />
      <span
        class="filename"
      />
      <button
        aria-label="remove file"
        class="bg-transparent border-0 ms-2"
      >
        <i
          class="bi bi-x-circle x-circle-icon"
        />
      </button>
    </li>
    <li
      class="fileupload-list-item"
    >
      <i
        class="bi bi-check-lg check-icon me-2"
      />
      <span
        class="filename"
        id="item"
      >
        item
      </span>
      <button
        aria-describedby="item"
        aria-label="remove file"
        class="bg-transparent border-0 ms-2"
      >
        <i
          class="bi bi-x-circle x-circle-icon"
        />
      </button>
    </li>
  </ul>
</DocumentFragment>
`;
